src/rating/rating.js

```diff
@@ -12,6 +12,13 @@ angular.module('ui.bootstrap.rating', [])
   this.init = function(ngModelCtrl_) {
     ngModelCtrl = ngModelCtrl_;
     ngModelCtrl.$render = this.render;
+    
+    ngModelCtrl.$formatters.push(function(value) {
+      if (angular.isNumber(value) && value % 1 !== 0) {
```

Comment:
`-1 % 1 !== 0` How about doing integer truncation so it'll be `value << 0 === value` instead?